common/src/main/java/com/wynntils/features/user/CombatXPGainMessageFeature.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.notifications.NotificationManager;
+import com.wynntils.mc.event.ClientTickEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.CharacterManager;
+import com.wynntils.wynn.utils.WynnUtils;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class CombatXPGainMessageFeature extends UserFeature {
+
+    @Config
+    public int tickDelay = 20;
+
+    private long tickCounter = 0;
+
+    private float newTickXP = 0;
+    private float lastTickXP = 0;
+    private float trackedPercentage = 0;
+    private int trackedLevel = 0;
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        tickCounter = 0;
+        newTickXP = 0;
+        lastTickXP = 0;
+        trackedPercentage = 0;
+    }
+
+    @SubscribeEvent
+    public void onTick(ClientTickEvent.End event) {
+        if (!WynnUtils.onWorld()) return;
+
+        tickCounter++;
+
+        if (tickCounter % tickDelay != 0) return;
+
+        tickCounter = 0;
+
+        gatherAndDispatchMessage();
+    }
+
+    private void gatherAndDispatchMessage() {
+        CharacterManager.CharacterInfo data = WynnUtils.getCharacterInfo();
+
+        int newLevel = data.getXpLevel();
+
+        // You get division by zero errors when you're at the level cap (i.e. 106 in Wynncraft 2.0.1).
+        // This needs to be updated if the level cap is ever raised.
+        if (newLevel > 105) return;
```

Comment:
I do not like this at all. It is one thing that it does not work on new versions, but it would be nice if we did not straight up crash the thing... The only reason why I accept this is that we do not crash since the feature will capture the crash and disable itself.